datasets/ar/parliament/crawler.py

```diff
@@ -0,0 +1,158 @@
+    if url_extension:
+        person_data["url"] = context.dataset.data.url + url_extension[
+            0
+        ].strip().replace("/diputados/", "")
+        crawl_personal_page(context, person_data)
+    else:
+        person_data.update(
+            {
+                "url": context.dataset.data.url,
+                "profession": None,
+                "birth_date": None,
+                "email": None,
+            }
+        )
+
+    # Extract additional available information about the person
+    person_data["district"] = _extract_text(element, ".//td[3]/text()")
+    person_data["term"] = _extract_text(element, ".//td[4]/text()")
+    person_data["term_start"] = _extract_text(element, ".//td[5]/text()")
+    person_data["term_end"] = _extract_text(element, ".//td[6]/text()")
+    person_data["block"] = _extract_text(element, ".//td[7]/text()")
+
+    context.log.info(f"Finished crawl person", person_data)
+    return person_data
+
+
+def _extract_text(element, xpath_query):
+    """
+    Helper function to extract text content from an HTML element using XPath.
+    Returns the extracted text, or None if not found.
+    """
+    result = element.xpath(xpath_query)
+    return result[0].strip() if result else None
+
+
+def crawl_personal_page(context: Context, person_data: dict):
+    """
+    Crawls a person's individual page and updates their data dictionary.
+    """
+    context.log.info("Starting crawling personal page", person_data["url"])
+    response = context.http.get(person_data["url"])
+    doc = html.fromstring(response.text)
+
+    # Extract additional details from the personal page
+    person_data["profession"] = _extract_text(
+        doc, './/p[@class="encabezadoProfesion"]/span/text()'
+    )
+    birth_date = _extract_text(doc, './/p[@class="encabezadoFecha"]/span/text()')
+    person_data["birth_date"] = convert_date_format(birth_date) if birth_date else None
+    person_data["email"] = _extract_text(
+        doc, './/a[starts-with(@href, "mailto:")]/text()'
+    )
+
+
+def make_and_emit_person(context: Context, person_data: dict):
+    """
+    Creates a 'Person' entity from the scraped data and emits it to the Context.
+    """
+    if person_data.get("name"):
+        person = context.make("Person")
+        person.id = context.make_id(person_data["name"])
+        person.add("name", person_data["name"])
+        person.add("alias", person_data["alias"])
+        person.add("country", "ar")
+        h.apply_name(
+            person,
+            first_name=person_data["first_name"],
+            last_name=person_data["last_name"],
+        )
+        person.add("sourceUrl", person_data["url"])
+        person.add("political", person_data["block"])
+
+        # Create position and occupancy
+        position = h.make_position(
+            context,
+            name="Member of National Congress of Argentina",
+            country="ar",
+            subnational_area=person_data.get("district"),
```

Comment:
Am I right in thinking district is their consitituency, rather than their jurisdiction? In that case, I think it's best not to put district in subnational_area, since it's intended to describe their jurisdiction. I totally understand the confusion though.

Reply:
I'll do that in a quick push in a bit